packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts

```diff
@@ -59,6 +61,10 @@ export const useScaffoldEventHistory = <
         throw new Error("Contract not found");
       }
 
+      if (!enabled) {
+        throw new Error("Hook disabled");
```

Comment:
Thanks for update!

But I think it shouldn't be error. If I assign `enabled = false` I'm expecting that there's no data and there's no error. I think 
```ts
if (!enabled) {
  return;
}
```
is ok.

Because let's say you handle errors in your react component

```ts
const {data, error} = useScaffoldEventHistory(...);

if (error) {
  return <p className='text-red-500'>{error}</div>
}

return {data?.map(...)}

```
Your example when filters is empty until query is parsed will show error first and then error will disappear. I think it's not good

Reply:
Yeah! Makes sense. Thanks!

Maybe we need:

```
if (!enabled) {
  setEvents(undefined);
  return;
}
```

To clean up the previous events when enabled true is swtiched to enabled false, right?

Reply:
Not sure, because if you remove events you also need to watch `fromBlockUpdated` and update it accordingly. If I understand right, returning `undefined` is better. For example in case when your `enabled` becomes `true`->`false` -> `true`, hook just adds events from last `fromBlockUpdated` and not reloading everything.

Reply:
I added this https://github.com/scaffold-eth/scaffold-eth-2/pull/621/files#diff-0efe8962fd8674c8a04310503b57e5e0ff36eb7016ae11764e1cc83472bc7646R122 to reload all the events when enabled change. I think that makes sense that if the hook is disabled, events is undefined.

Reply:
you're right, approve!